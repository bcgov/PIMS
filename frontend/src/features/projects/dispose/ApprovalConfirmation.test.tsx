import React from 'react';
import renderer from 'react-test-renderer';
import { Formik } from 'formik';
import { Form } from 'react-bootstrap';
import { noop } from 'lodash';

const renderComponent = (checked: boolean) => {
  //const userDisplayName = 'PIMS User';
  const fieldName = 'confirmation';
  return renderer.create(
    <Formik initialValues={{ [fieldName]: checked }} onSubmit={() => {}}>
      <Form>
        {/* <ApprovalConfirmationCheckbox
          field={fieldName}
          onChange={onChange}
          userDisplayName={userDisplayName}
        /> */}
      </Form>
    </Formik>,
  );
};

describe('Approval Confirmation', () => {
  xit('Matches Snapshot', () => {
    const component = renderComponent(false, noop);
    expect(component.toJSON()).toMatchSnapshot();
  });

  //TODO: update after form refactor.
  xit('Input is checked', () => {
    const component = renderComponent(true, noop);
    expect(component.root.findByType('input').props.checked).toBeTruthy();
  });

  xit('Input is not checked', () => {
    const component = renderComponent(false, noop);
    expect(component.root.findByType('input').props.checked).toBeFalsy();
  });
});
