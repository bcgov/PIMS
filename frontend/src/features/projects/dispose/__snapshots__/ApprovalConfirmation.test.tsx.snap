// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Approval Confirmation Matches Snapshot 1`] = `
<form
  className=""
>
  <div
    className="approvalGroup form-group"
  >
    <div
      className="form-row"
    >
      <input
        className="col-md-1"
        name="confirmation"
        onBlur={[Function]}
        onChange={[Function]}
        type="checkbox"
        value={false}
      />
      <label
        className="col-md-10 form-label col-form-label col"
        htmlFor="approval-confirmation"
      >
        I, PIMS User, confirm by checking this box that I have received the approval 
    to request the property be submitted to the Enhanced Referral Program and marketed internally to other Ministry 
    and Broader Public Sector Agencies for 90 days before listing the property to be sold on the public commercial real estate market
      </label>
    </div>
  </div>
</form>
`;
